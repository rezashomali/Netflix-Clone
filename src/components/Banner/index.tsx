import React, { useState, useEffect } from "react";
import axios from "../../Api/axios";
import requests from "../../Api/requests";
import "./banner.css";

interface MovieType {
  backdrop_path: string;
  name?: string;
  title?: string;
  original_name?: string;
  overview: string;
}

const Banner: React.FC = () => {
  const [movie, setMovie] = useState<MovieType>();

  useEffect(() => {
    async function fetchData() {
      const request = await axios.get(requests.fetchNetflixOriginals);
      setMovie(
        request.data.results[
          Math.floor(Math.random() * request.data.results.length - 1)
        ]
      );
    }
    fetchData();
  }, []);

  const truncate: Function = (str: string, n: number) => {
    return str?.length > n ? str.substr(0, n - 1) + "..." : str;
  };

  return (
    <header
      className="banner"
      style={{
        backgroundSize: "cover",
        backgroundImage: `url("https://image.tmdb.org/t/p/original/${movie?.backdrop_path}")`,
        backgroundPosition: "top center",
      }}
    >
      <div className="banner__contents">
        <h1 className="banner__title">
          {movie?.name || movie?.title || movie?.original_name}
        </h1>
        <div className="banner__buttons">
          <button className="banner__button">Play</button>
          <button className="banner__button">My List</button>
        </div>
        <p className="banner__description">{truncate(movie?.overview, 150)}</p>
      </div>
      <div className="banner--fadeBottom" />
    </header>
  );
};

export default Banner;
