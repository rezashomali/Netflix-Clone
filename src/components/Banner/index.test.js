import React from "react";
import renderer from "react-test-renderer";
import Banner from "./index";

test("Nav snapshot ", () => {
  const component = renderer.create(<Banner />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
