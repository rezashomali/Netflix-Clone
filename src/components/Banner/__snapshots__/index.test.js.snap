// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot  1`] = `
<header
  className="banner"
  style={
    Object {
      "backgroundImage": "url(\\"https://image.tmdb.org/t/p/original/undefined\\")",
      "backgroundPosition": "top center",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="banner__contents"
  >
    <h1
      className="banner__title"
    />
    <div
      className="banner__buttons"
    >
      <button
        className="banner__button"
      >
        Play
      </button>
      <button
        className="banner__button"
      >
        My List
      </button>
    </div>
    <p
      className="banner__description"
    />
  </div>
  <div
    className="banner--fadeBottom"
  />
</header>
`;
